packages/atclient/include/atclient/socket.h

```diff
@@ -0,0 +1,113 @@
+#ifndef ATCLIENT_SOCKET_H
+#define ATCLIENT_SOCKET_H
+#include <atclient/socket_shared.h>
+#include <stddef.h>
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(ATCLIENT_SOCKET_PROVIDER_EXTERNAL)
+// Noop, this indicates an external socket provider will be linked
+#else
+#define ATCLIENT_SOCKET_PROVIDER_MBEDTLS
+#endif
+
+// IWYU pragma: begin_exports
+
+// Export the appropriate platform specific struct implementation
+#if defined(ATCLIENT_SOCKET_PROVIDER_MBEDTLS)
+#include "socket_mbedtls.h"
+#endif
+
+// IWYU pragma: end_exports
+
+/**
+ * @brief Initializes a raw socket
+ *
+ * @param socket The socket structure to initialize
+ */
+void atclient_raw_socket_init(struct atclient_raw_socket *socket);
+
+/**
+ * @brief Frees resources associated with a network socket
+ *
+ * @param socket The socket structure to free resources from
+ */
+void atclient_raw_socket_free(struct atclient_raw_socket *socket);
```

Comment:
Raw sockets only have init and free for now. I plan to add more functionality later, but all we need is init and free right now, as the ssl context takes ownership of the object.